elorus/client.py

```diff
@@ -0,0 +1,187 @@
+from dataclasses import asdict
+from typing import Optional
+
+import httpx
+
+from elorus.auth import ElorusAuthentication
+from elorus.exceptions import (
+    AuthenticationError,
+    AuthorizationError,
+    BadRequestError,
+    Error,
+    ThrottlingError,
+)
+from elorus.models import Contact, EmailBody, Invoice
+
+
+class Client:
+    def __init__(
+        self,
+        api_key: str,
+        elorus_organization_id: str,
+        is_demo: bool = False,
+        base_url: Optional[str] = "https://api.elorus.com",
+        api_version: Optional[str] = "v1.1",
+    ):
+        self.base_url = base_url
+        self.api_key = api_key
+        self.elorus_organization_id = elorus_organization_id
+        self.is_demo = is_demo
+        self.api_version = api_version
+
+        self.contacts = Contacts(self)
+        self.invoices = Invoices(self)
+
+    def _get_auth(self):
+        return ElorusAuthentication(
+            token=self.api_key,
+            elorus_organization_id=self.elorus_organization_id,
+            is_demo=self.is_demo,
+        )
+
+    def handle_file_download(self, response):
+        filename = response.headers.get("Content-Disposition", None)
+        if filename:
+            filename = filename.split("filename=")[1]
+        return filename, response.content
```

Comment:
I suggest we simplify this a bit and also use better naming. Example:

```suggestion
        content_disposition = response.headers.get("Content-Disposition", "")
        filename_kv = filename.split("filename=")
        if len(filename_kv) >= 2:
            filename = filename_kv.split("filename=")[1]
            return filename, response.content
```